### ISSUE TITLE
[Suggestion] Proposed station connector redesign.

### ISSUE THREAD

**Issue by Womble-TC** [tsunamayo/Starship-EVO#2289]
While the current station connector tech is really cool, its currently very hard to integrate into any design due the entrance  being bulky and round:
![image](https://user-images.githubusercontent.com/29961381/83362615-93e08280-a38a-11ea-9e09-acc1c175001a.png) For me at least, this is making it quite difficult to build any station modules.

To maybe rectify this, I have come up with some tweaks that could be made to the design.

- Have the "interior" part of the station converter be flat, so it can easily be made part of a room or corridor without taking up lots of space: 
![image](https://user-images.githubusercontent.com/29961381/83362658-e6ba3a00-a38a-11ea-9c67-48895d528812.png) (This door could also be round, like the exterior door, but I'll get to that later.

-Here, I added some space in the inside of the docking tube so that a player could add details: 
![image](https://user-images.githubusercontent.com/29961381/83362691-1bc68c80-a38b-11ea-8591-c2cc483ce063.png)
The indents are 0.5m in depth, so people could easily add some nice pipes and stuff if they wanted to.

-Finally, I thought it would be cool to have a fancy door on the outside of the converter. This iris door wouldn't clip into anything it is connected too: 
![door_gif](https://user-images.githubusercontent.com/29961381/83362768-a909e100-a38b-11ea-8e3a-64193e0685ed.gif)

There are some changes that could be made, but I think this would generally be and improvement, and make building stations using these blocks a good bit easier.

Here is the blender file if you want to look at it for yourself: 
[connecter_2.zip](https://github.com/tsunamayo/Starship-EVO/files/4708478/connecter_2.zip)

**Comment by tsunamayo** [comment]
I like it! Could you send me the file in fbx? Thanks

**Comment by Womble-TC** [comment]
sure, one sec

**Comment by Womble-TC** [comment]
[fbx.zip](https://github.com/tsunamayo/Starship-EVO/files/4708663/fbx.zip)
Not sure if it will work since the model has an animation, and is in separate bits. I put the blend file earlier if that helps too.

**Comment by tsunamayo** [comment]
yeah I will have to code something for the animation anyway ;) Worse case I can redo it on my own no pb. Does the blade of the iris travel in a perpendicular way to their bisector? Like that: https://www.instructables.com/id/Re-engineered-Iris-Diaphragm/

**Comment by Womble-TC** [comment]
Yes they do (or they are supposed to at least)

**Comment by TripleHelixxx** [comment]
I’d like to add to this suggestion if I may.
I’ve been playing around with the new station tech, and the connector hatches look fine, though  I do kind of like the design shown above more. The problem I do have with them is that they’re tedious to traverse and generally just not fun to deal with when moving from module to module. 
As it stands right now if you want to move from one module to the next, you have to first open the connectors inner hatch, then get inside and close that hatch, then open that connectors outer hatch as well as the next modules connectors outer hatch, then move to the next connector, then close both outer hatches, then finally open the inner hatch. As you can imagine, this could get very annoying if you’re constantly having to move back-and-forth between modules for some reason. 

My solution, a second simpler connector, used only for connecting modules and traversing between them. It would only have one door, so you wouldn’t be able to use it in the airlock like fashion of the current connector to get in and out of a station, only go from module to module. I’m not even sure if this would be possible, but I feel like people would really like it.

**Comment by tsunamayo** [comment]
@TripleHelixxx yes there is clearly a door fatigue, it was also my feeling when trying the system a bit.
My plan to address that:
1) You dont need to close the first door, it will auto-close when you open the second door.
2) when two connector are docked, the door are opened all the time!

This should significantly reduce the door fatigue! 
I could even get an auto close / auto second door open if that is necessary, but we will see after the first two improvements are implemented.

**Comment by tsunamayo** [comment]
okay that design looks cool on paper, but for a 2m wide dock it wont work as the opening can only be 1m... Thats just too tight...

**Comment by Womble-TC** [comment]
maybe have it 3x3 with a 2m hole? it would be a tight squeeze but I can figure something out if you wanted me to try. I would texture it too, but I'm not sure what your workflow is (having parts of the model printable and such).

**Comment by Womble-TC** [comment]
There's a few benefits to having it 3x3 too, it means you could mount it on the front of a ship which is an odd number of blocks wide, and it would have enough space to detail the interior a bit too.

**Comment by Womble-TC** [comment]
@tsunamayo Here is a redesign for a possible 3x3 connector. I think a 3x3 design would work better for the reasons stated above (and mostly because you cant have the cool door with a 2x2 design). 
![door 2](https://user-images.githubusercontent.com/29961381/83572593-c5d81d00-a521-11ea-80db-4012ea937c35.gif)
The opening hole is 2m in diameter (but with the iris door, its decreases this to 1.75m). The interior would still have 0.25m indents to allow players to detail their docking ports on their ships. 
![image](https://user-images.githubusercontent.com/29961381/83572771-118ac680-a522-11ea-9ed0-3342e93bc9aa.png)
![image](https://user-images.githubusercontent.com/29961381/83572887-472faf80-a522-11ea-8927-518cf3ffafbe.png)

**Comment by PopkaAkula** [comment]
Wow womble great job, this looks really cool! :)

**Comment by Creepthefrenchpastery** [comment]
Cool :)

**Comment by tsunamayo** [comment]
Yes it looks fantastique, but I already did mine, a 3m wide with an iris. The hook mechanism is very cool, but I cant reproduce it with my tech...
Cheers

**Comment by Womble-TC** [comment]
No problem at all! I like how your one looks anyway.
